vllm/spec_decode/draft_model_runner.py

```diff
@@ -132,35 +225,37 @@ def execute_model(
         if not self.is_driver_worker:
             raise ValueError("TP1DraftModelRunner only supports TP=1.")
 
-        if self.lora_config:
-            assert model_input.lora_requests is not None
-            assert model_input.lora_mapping is not None
-            self.set_active_loras(model_input.lora_requests,
-                                  model_input.lora_mapping)
+        # Sanity
+        assert self.lora_config is None
+        assert self.prompt_adapter_config is None
+        assert model_input.attn_metadata is not None
 
-        if self.prompt_adapter_config:
-            assert model_input.prompt_adapter_requests is not None
-            assert model_input.prompt_adapter_mapping is not None
-            self.set_active_prompt_adapters(
-                model_input.prompt_adapter_requests,
-                model_input.prompt_adapter_mapping)
+        # Detect exec mode
+        use_cuda_graph = False
+        if model_input.attn_metadata.num_prefills > 0:
+            # In this case, execute_model(..) was called directly
+            assert num_steps == 1, (
+                "execute_model(..) of draft_model_runner can be called "
+                "directly only with a single-step prefill")
+        else:
+            use_cuda_graph = model_input.attn_metadata.use_cuda_graph
+            # We can skip CPU samples for spec token generation
+            model_input.sampling_metadata.skip_sampler_cpu_output = True
```

Comment:
It does allow single-prefill run because it is used in warmup (and it bypasses the supports_gpu_multi_step(..) function). But I have added the exception.